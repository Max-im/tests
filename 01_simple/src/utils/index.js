export const getNexId = arr => {
  let maxId = 0;
  arr.forEach(item => {
    if (maxId < item.id) maxId = item.id;
  });
  return [{ id: maxId + 1 }, ...arr];
};
