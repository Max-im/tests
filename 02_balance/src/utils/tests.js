import React from "react";
import renderer from "react-test-renderer";

export const isRender = component => {
  const c = renderer.create(component);
  expect(c.toJSON()).toMatchSnapshot();
};
