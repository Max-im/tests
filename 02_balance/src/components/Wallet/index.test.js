import React from "react";
import { shallow } from "enzyme";
import { Wallet } from "./";
import { isRender } from "../../utils/tests";

describe("Wallet", () => {
  const mockDeposit = jest.fn();
  const mockWithdraw = jest.fn();

  const props = {
    balance: 20,
    onDeposit: mockDeposit,
    onWithdraw: mockWithdraw
  };
  const wallet = shallow(<Wallet {...props} />);
  test("component renders successful", () => isRender(wallet, "wallet"));
  test("title renders successful", () => isRender(wallet, "wallet__title"));
  test("input renders successful", () => isRender(wallet, "wallet__input"));
  test("deposit btn renders properly", () =>
    isRender(wallet, "wallet__depositBtn"));
  test("withdraw renders successful", () =>
    isRender(wallet, "wallet__withdrawBtn"));

  test("display balance from props", () => {
    expect(wallet.find(".balance").text()).toBe("Balance 20");
  });

  test("deposit and withdraw inputs exsist", () => {
    expect(wallet.find(".input-wallet").exists()).toBe(true);
  });

  describe("typing into the input", () => {
    const userBalance = "25";
    beforeEach(() =>
      wallet
        .find(".input-wallet")
        .simulate("change", { target: { value: userBalance } })
    );

    test("change user balance in state", () => {
      expect(wallet.state().balance).toBe(userBalance - 0);
    });

    describe("add deposit", () => {
      beforeEach(() => wallet.find(".btn-deposit").simulate("click"));

      test("deposit is called", () => {
        expect(mockDeposit).toHaveBeenCalledWith(parseInt(userBalance, 10));
      });
    });

    describe("withdraw event", () => {
      beforeEach(() => wallet.find(".btn-withdraw").simulate("click"));

      test("withdraw is called", () => {
        expect(mockWithdraw).toHaveBeenCalledWith(parseInt(userBalance, 10));
      });
    });
  });
});
