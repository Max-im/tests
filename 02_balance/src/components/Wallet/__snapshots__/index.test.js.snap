// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet render successful 1`] = `
<div>
  <h3
    className="balance"
  >
    Balance 
    20
  </h3>
  <br />
  <input
    className="input-wallet"
    onChange={[Function]}
  />
  <button
    className="btn-deposit"
    onClick={[Function]}
  >
    add deposit
  </button>
  <button
    className="btn-withdraw"
    onClick={[Function]}
  >
    withdraw
  </button>
</div>
`;
